pkg/agent/clean/orphan_bindings_test.go

```diff
@@ -0,0 +1,480 @@
+package clean
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/auth"
+)
+
+func TestCleanOrphans(t *testing.T) {
```

Comment:
Nit: personally speaking, I'm not a fan of splitting tests which test the same function into separate test functions. It can make it harder to track which tests are testing which functions.

You are also calling this function `TestCleanOrphans` but it's testing the non-exported function `cleanOrphans`, so really this should be something like `Test_cleanOrphans`.

Marking this as a nit since it's a structural test comment, and those are entirely implementer's choice.

Reply:
This is completely fine, even good. There are many examples in Go's stdlib (see tests of the `bufio` package, for example). We're testing behaviors, not necessarily functions. We're testing packages, I could even say.
The name of the test is then key - it's OK for it to be long but descriptive.

These are example tests from bufio_test.go:
TestNoUnreadByteAfterPeek
TestNoUnreadByteAfterDiscard

Having said that, I think I might want to name my test better, yes. And about `_` in names - there is no need in them at all. The test should pretty much always be camel case.

Reply:
I appreciate your argument, and the references to go's stdlib, but I will respectfully have to disagree with you here. 

> We're testing behaviors, not necessarily functions. We're testing packages, I could even say.

The exported values (including structs and functions) are the behaviors of a package. You test the behavior of a package by testing the contracts that it offers to consumers, which are these exported functions/types.

>  And about _ in names - there is no need in them at all. The test should pretty much always be camel case.

From what I can see here, you are testing a method on a type, which as per: https://pkg.go.dev/testing#hdr-Examples should be `TestT_M()` with an underscore. 

As said above, this is all implementer's choice, so we can resolve if you are ok with what you have implemented so far.

Reply:
I agree, but not sure how my testing setup contradicts this. I am testing the contract. It's just that the set of behaviors is split into multiple test cases. I still don't see how there is difficulty in tracking what's being tested, especially if test names are descriptive. They aren't even split into multiple files...

The link you provided talks about Example tests, a special kind of tests, which might have different conventions.

Reply:
It looks like there are no well-defined conventions, and the Examples test guidelines are special https://github.com/golang/go/issues/33688